File: src/main/java/com/clearcont/clearcontapp/helpers/CookieFactory.java
package com.clearcont.clearcontapp.helpers;

import com.vaadin.flow.server.VaadinResponse;
import com.vaadin.flow.server.VaadinService;
import jakarta.servlet.http.Cookie;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@AllArgsConstructor
@Slf4j
public class CookieFactory {
    VaadinResponse response;
    private final String CLASS_NAME = CookieFactory.class.getSimpleName();
    
    public void setCookie(String cookieName, String value) {
        Cookie cookie = new Cookie(cookieName, value);
        cookie.setMaxAge(60 * 60 * 24);
        response.addCookie(cookie);
    }
    
    public Long getCookieInteger(String cookieName) {
        long id = 0L;
        Cookie[] cookies = VaadinService.getCurrentRequest().getCookies();
        for (Cookie cookie : cookies) {
            if (cookie.getName().equals(cookieName)) {
                id = Long.parseLong(cookie.getValue());
                log.info(CLASS_NAME, "O valor do cookie é: " + cookie.getValue());
                break;
            }
        }
        return id;
    }
    
    public String getCookieString(String cookieName) {
        String cookieSaved = "";
        Cookie[] cookies = VaadinService.getCurrentRequest().getCookies();
        for (Cookie cookie : cookies) {
            if (cookie.getName().equals(cookieName)) {
                cookieSaved = cookie.getValue();
                log.info(CLASS_NAME, "O valor do cookie é: " + cookie.getValue());
                break;
            }
        }
        return cookieSaved;
    }
}

File: src/main/kotlin/br/com/clearcont/clearcontwebapp/helpers/CookieFactory.kt
package br.com.clearcont.clearcontwebapp.helpers

import com.vaadin.flow.server.VaadinResponse
import com.vaadin.flow.server.VaadinService
import jakarta.servlet.http.Cookie
import java.util.logging.Logger

class CookieFactory(
    private var response: VaadinResponse
) {
    private val log: Logger = Logger.getLogger(javaClass.name)

    fun setCookie(cookieName: String?, value: String?) {
        val cookie = Cookie(cookieName, value)
        cookie.maxAge = 60 * 60 * 24
        response.addCookie(cookie)
    }

    fun getCookieInteger(cookieName: String): Long {
        var id = 0L
        val cookies = VaadinService.getCurrentRequest().cookies
        for (cookie in cookies) {
            if (cookie.name == cookieName) {
                id = cookie.value.toLong()
                log.info("O valor do cookie é:  ${cookie.value}")
                break
            }
        }
        return id
    }

    fun getCookieString(cookieName: String): String {
        var cookieSaved = ""
        val cookies = VaadinService.getCurrentRequest().cookies
        for (cookie in cookies) {
            if (cookie.name == cookieName) {
                cookieSaved = cookie.value
                log.info("O valor do cookie é:  ${cookie.value}")
                break
            }
        }
        return cookieSaved
    }
}
